src/buffer/out/textBuffer.cpp

```diff
@@ -346,6 +346,11 @@ OutputCellIterator TextBuffer::Write(const OutputCellIterator givenIt,
         ++lineTarget.Y;
     }
 
+    // Take the cell distance written and notify that it needs to be repainted.
+    const auto written = it.GetCellDistance(givenIt);
+    const Viewport paint = Viewport::FromDimensions(target, { gsl::narrow<SHORT>(written), 1 });
+    _NotifyPaint(paint);

```

Comment:
Won't this _not_ work if the write wraps to the next line? This looks to me like it'll only invalidate the region of the buffer on the initial line, and none of the subsequent lines.